### DISCUSSION TITLE
How to make it work with electron & webpack serve

### DISCUSSION THREAD

**Discussion by tiagorp** [suren-atoyan/monaco-react#196] [Q&A]
Hi,

I'm trying to use monaco-react together with electron and webpack but so far without much success. Would really appreciate some help.

My current project setup has been created from scratch, that is, without any boilerplate such as CRA.

In order to test my electron app, I use webpack serve. I'm struggling in configuring the monaco-react to use my node_modules monaco-editor package instead of the CDN.

Since I'm using webpack serve, electron runs the window from localhost:4000/ (which is what I've configured in my webpack config file).

In order to be able to use monaco-editor locally instead of the CDN, I've:
1. Installed monaco-editor as a (node_modules) dependency.
2. Installed the CopyWebpackPlugin so I can tell webpack to copy the the '/node_modules/monaco-editor/min/vs' files to my dist folder (see code below).

`new CopyWebpackPlugin({
	patterns: [
		{
			from: path.resolve(__dirname, '../node_modules', 'monaco-editor', 'min', 'vs'),
			to: 'vs',
		},
		{
			from: path.resolve(__dirname,'../node_modules','monaco-editor','min-maps','vs'),
			to: path.join('min-maps/vs'),
		},
	],
})`

This part seems to be working as when I fire up webpack dev server I can see a vs folder on my localhost:4000 together with my bundle.js.

Now the problem is how do I tell my app that the vs folder is actually at my localhost:4000/vs instead of the CDN?

I've tried the following:

`    loader.config({
        paths: { vs: 'http://localhost:4000/vs' },
    });`

However this results in the following error: 

![image](https://user-images.githubusercontent.com/14852789/109436037-73c31a00-7a15-11eb-8872-dc98422a688a.png)

I've tried already other paths but so far no success, so any ideas are welcome.

Thanks,
Tiago

**Comment by tiagorp** [comment]
I've tried with a simpler customer electron boilerplate and it seems the problem might not be related to the devpack webserver but to the ability of the Monaco initializing the editor main js.

I still haven't figured it out.

**Comment by tiagorp** [comment]
Related issue and solution [here](https://github.com/suren-atoyan/monaco-react/issues/203)

**Comment by hekar** [comment]
I've the same issue, but I'm attempting to use `electron-serve` to serve the file from `app://./resources/vs`

**Reply by hekar** [reply]
Reproduced the same issue using [Nextron](https://github.com/saltyshiomix/nextron/) (uses NextJS.)

```
Error: ENOENT: no such file or directory, open 'C:\Users\hr\code\descriptum\http:\localhost:8888\vs\editor\editor.main.js'
```

**Reply by tiagorp** [reply]
Hi, @hekar .

Does the solution described in this [link](https://github.com/suren-atoyan/monaco-react/issues/203) helped you?

**Comment by xmsz** [comment]
any solution?
